AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m

```diff
@@ -374,6 +380,20 @@ - (void)sendLog:(id<MSLog>)log flags:(MSFlags)flags {
   [self.channelUnit enqueueItem:log flags:flags];
 }
 
+- (void)setTransmissionInterval:(NSUInteger)interval {
+  if (interval > kMSFlushIntervalMaximum || interval < kMSFlushIntervalMinimum) {
+    MSLogWarning([MSAnalytics logTag],
+                 @"The transmission interval is invalid, it should be between 3 second and 1 day (86400 seconds).");
+    return;
+  }
+  if (self.started) {
+    MSLogWarning([MSAnalytics logTag], @"The transmission interval should be set before the MSAnalytics service is started.");
+    return;
+  }
+  self.flushInterval = interval;
+  self.channelUnitConfiguration = [[MSChannelUnitConfiguration alloc] initWithGroupId:[self groupId] flushInterval:self.flushInterval];
```

Comment:
I don't think this is a good idea to re-instantiate channel unit configuration.
Remove instantiation in `init` and do this when Analytics is started.